dbt/include/duckdb/macros/materializations/external.sql

```diff
@@ -2,9 +2,22 @@
 
   {%- set location = render(config.get('location', default=external_location(this, config))) -%})
   {%- set rendered_options = render_write_options(config) -%}
-  {%- set format = config.get('format', 'parquet') -%}
+
+  {%- set format = config.get('format') -%}
+  {%- set allowed_formats = ['csv', 'parquet', 'json'] -%}
+
+  {%- if format -%}
+      {%- set format = format if format in allowed_formats else 'parquet' -%}
+  {%- else -%}
+      {%- set format = location.split('.')[-1] if '.' in location else 'parquet' -%}
```

Comment:
we might also want to do a lower on the format string here so we don't fail for `.CSV` or `.JSON` (or worse, silently write out parquet instead of the requested format)